### ISSUE TITLE
Error running script

### ISSUE THREAD

**Issue by galaxy-stargraph** [sj-dan/OpenSea-NFT-Stealer#9]
I get this error when trying to run this script
`Traceback (most recent call last):
  File "C:\Users\17379\Desktop\OpenSea-NFT-Stealer-master\opensea.py", line 60, in <module>
    data = json.loads(requests.get(f"https://api.opensea.io/api/v1/assets?order_direction=asc&offset={offset}&limit=50&collection={CollectionName}&format=json").content.decode())
  File "C:\Users\17379\AppData\Local\Programs\Python\Python39\lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "C:\Users\17379\AppData\Local\Programs\Python\Python39\lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Users\17379\AppData\Local\Programs\Python\Python39\lib\json\decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)`

**Comment by NtekShadow** [comment]
first define headers with a user client like this:

```
  headers = {
      'User-Agent': 'Put the userclient you want to use in here'
  }
```
you can find a userclient name on here: https://www.whatismybrowser.com/guides/the-latest-user-agent/

then change the request.get in the first for loop for data to include the headers:

```
data = json.loads(requests.get(
                f"https://api.opensea.io/api/v1/assets?order_direction=asc&offset={offset}"
                f"&limit=50&collection={CollectionName}&format=json", headers=headers).content.decode())
```
and it should work again.

**Comment by Salro1g** [comment]
Thanks dude!!! Nice work!

**Comment by samuelbailey123** [comment]
Running into a new error.... 
Perhaps my header is incorrect...?

#0000:
  Data  -> [✓] (Already Downloaded)
Traceback (most recent call last):
  File "/Users/samuel.bailey/Desktop/NFT/OpenSea-NFT-Stealer/opensea.py", line 88, in <module>
    image = requests.get(asset["image_original_url"])
  File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/sessions.py", line 649, in send
    adapter = self.get_adapter(url=request.url)
  File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/sessions.py", line 742, in get_adapter
    raise InvalidSchema("No connection adapters were found for {!r}".format(url))
requests.exceptions.InvalidSchema: No connection adapters were found for 'ipfs://QmRRPWG96cmgTn2qSzjwr2qvfNEuhunv6FNeMFGa9bx6mQ

Also this error is only on some collections.

Update: 
Seems like all of the ape collections end in this error.

**Comment by samuelbailey123** [comment]
also curious if anyone knows how to find the actual collection names?

**Comment by NtekShadow** [comment]
@samuelbailey123 
```
headers = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36'
}
```

try this one, works for me with ape collections

**Comment by samuelbailey123** [comment]
Still no luck @NtekShadow 
I am on mac, everyone online is using a windows for ape but I'm not understanding why that would matter?

**Comment by NtekShadow** [comment]
no idea, i just noticed that the request with user client works, i didnt test any other.

**Comment by megapro17** [comment]
@NtekShadow Bruh just upload the fixed script

**Comment by samuelbailey123** [comment]
@NtekShadow PR to for headers
https://github.com/sj-dan/OpenSea-NFT-Stealer/pull/11

**Comment by samuelbailey123** [comment]
https://github.com/sj-dan/OpenSea-NFT-Stealer/pull/11

**Comment by Salro1g** [comment]
Following NtekShadow instructions is working well on windows and MacOS for me.

**Comment by NtekShadow** [comment]
@megapro17 bro i didnt write this script, you can wait for a response of the og creator if you like that more...

**Comment by galaxy-stargraph** [comment]
> first define headers with a user client like this:
> 
> ```
>   headers = {
>       'User-Agent': 'Put the userclient you want to use in here'
>   }
> ```
> 
> you can find a userclient name on here: https://www.whatismybrowser.com/guides/the-latest-user-agent/
> 
> then change the request.get in the first for loop for data to include the headers:
> 
> ```
> data = json.loads(requests.get(
>                 f"https://api.opensea.io/api/v1/assets?order_direction=asc&offset={offset}"
>                 f"&limit=50&collection={CollectionName}&format=json", headers=headers).content.decode())
> ```
> 
> and it should work again.

Thanks so much mate!

**Comment by zhincic** [comment]
I have tried working on fixing this, but seems like I am not so good at it. Is there anyone who can upload the opensea.py file that works or screenshot the script that works?
<img width="407" alt="Screenshot 2022-01-18 at 06 28 49" src="https://user-images.githubusercontent.com/90480378/149876484-1ab16fca-d5b8-457e-b2de-2d9b256c6c8c.png">

**Comment by hyper10110** [comment]
> Running into a new error.... Perhaps my header is incorrect...?
> 
> #0: Data -> [✓] (Already Downloaded) Traceback (most recent call last): File "/Users/samuel.bailey/Desktop/NFT/OpenSea-NFT-Stealer/opensea.py", line 88, in image = requests.get(asset["image_original_url"]) File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/api.py", line 75, in get return request('get', url, params=params, **kwargs) File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/api.py", line 61, in request return session.request(method=method, url=url, **kwargs) File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/sessions.py", line 542, in request resp = self.send(prep, **send_kwargs) File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/sessions.py", line 649, in send adapter = self.get_adapter(url=request.url) File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/sessions.py", line 742, in get_adapter raise InvalidSchema("No connection adapters were found for {!r}".format(url)) requests.exceptions.InvalidSchema: No connection adapters were found for 'ipfs://QmRRPWG96cmgTn2qSzjwr2qvfNEuhunv6FNeMFGa9bx6mQ
> 
> Also this error is only on some collections.
> 
> Update: Seems like all of the ape collections end in this error.

Yeah im gettiing similar error in a different collection, Gemma

#67:
  Image -> [✓] (Already Downloaded)

#68:
Traceback (most recent call last):
  File "E:\OpenSea-NFT-Stealer-master\OpenSea-NFT-Stealer-master\opensea.py", line 98, in <module>
    image = requests.get(asset["image_original_url"])
  File "C:\Users\stuar\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "C:\Users\stuar\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\Users\stuar\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\sessions.py", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\Users\stuar\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\sessions.py", line 639, in send
    adapter = self.get_adapter(url=request.url)
  File "C:\Users\stuar\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\sessions.py", line 732, in get_adapter
    raise InvalidSchema("No connection adapters were found for {!r}".format(url))
requests.exceptions.InvalidSchema: No connection adapters were found for 'ipfs://QmUjGTEirKvrLhdjthpHxMD91WkzJ8vjsZCSCo89RmqyCC/d8d2052b6b50e809c1fda3b12c09ab1ec2cfc2eaa4b1bdf73eae6c1ec8248d82.jpg'

**Comment by samuelbailey123** [comment]
> I have tried working on fixing this, but seems like I am not so good at it. Is there anyone who can upload the opensea.py file that works or screenshot the script that works? <img alt="Screenshot 2022-01-18 at 06 28 49" width="407" src="https://user-images.githubusercontent.com/90480378/149876484-1ab16fca-d5b8-457e-b2de-2d9b256c6c8c.png">

This error means you don't have the module installed on your computer 
try installing it with pip

**Comment by samuelbailey123** [comment]
> > Running into a new error.... Perhaps my header is incorrect...?
> > #0: Data -> [✓] (Already Downloaded) Traceback (most recent call last): File "/Users/samuel.bailey/Desktop/NFT/OpenSea-NFT-Stealer/opensea.py", line 88, in image = requests.get(asset["image_original_url"]) File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/api.py", line 75, in get return request('get', url, params=params, **kwargs) File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/api.py", line 61, in request return session.request(method=method, url=url, **kwargs) File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/sessions.py", line 542, in request resp = self.send(prep, **send_kwargs) File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/sessions.py", line 649, in send adapter = self.get_adapter(url=request.url) File "/Users/samuel.bailey/.pyenv/versions/3.9.7/lib/python3.9/site-packages/requests/sessions.py", line 742, in get_adapter raise InvalidSchema("No connection adapters were found for {!r}".format(url)) requests.exceptions.InvalidSchema: No connection adapters were found for 'ipfs://QmRRPWG96cmgTn2qSzjwr2qvfNEuhunv6FNeMFGa9bx6mQ
> > Also this error is only on some collections.
> > Update: Seems like all of the ape collections end in this error.
> 
> Yeah im gettiing similar error in a different collection, Gemma
> 
> #67: Image -> [✓] (Already Downloaded)
> 
> #68: Traceback (most recent call last): File "E:\OpenSea-NFT-Stealer-master\OpenSea-NFT-Stealer-master\opensea.py", line 98, in image = requests.get(asset["image_original_url"]) File "C:\Users\stuar\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\api.py", line 75, in get return request('get', url, params=params, **kwargs) File "C:\Users\stuar\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\api.py", line 61, in request return session.request(method=method, url=url, **kwargs) File "C:\Users\stuar\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\sessions.py", line 529, in request resp = self.send(prep, **send_kwargs) File "C:\Users\stuar\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\sessions.py", line 639, in send adapter = self.get_adapter(url=request.url) File "C:\Users\stuar\AppData\Local\Programs\Python\Python39\lib\site-packages\requests\sessions.py", line 732, in get_adapter raise InvalidSchema("No connection adapters were found for {!r}".format(url)) requests.exceptions.InvalidSchema: No connection adapters were found for 'ipfs://QmUjGTEirKvrLhdjthpHxMD91WkzJ8vjsZCSCo89RmqyCC/d8d2052b6b50e809c1fda3b12c09ab1ec2cfc2eaa4b1bdf73eae6c1ec8248d82.jpg'

Still working on a fix for this

**Comment by danman138** [comment]
Was working last night for me with the fix now i cant even download same project. # Define variables for statistics
headers = {
      'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 12_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36'
  }
stats = {
"DownloadedData": 0,
"AlreadyDownloadedData": 0,
"DownloadedImages": 0,
"AlreadyDownloadedImages": 0,
"FailedImages": 0
}

# Iterate through every unit
for i in range(iter):
    offset = i * 50
    data = json.loads(requests.get(
                f"https://api.opensea.io/api/v1/assets?order_direction=asc&offset={offset}"
                f"&limit=50&collection={CollectionName}&format=json", headers=headers).content.decode())

    if "assets" in data:
        for asset in data["assets"]:
          formatted_number = f"{int(asset['token_id']):04d}"

**Comment by danman138** [comment]
Worked last night with @NtekShadow 's fix now im getting same error again. "Traceback (most recent call last):"  lines 62 346 337 355
```
# Define variables for statistics
headers = {
'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 12_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36'
}
stats = {
"DownloadedData": 0,
"AlreadyDownloadedData": 0,
"DownloadedImages": 0,
"AlreadyDownloadedImages": 0,
"FailedImages": 0
}

# Iterate through every unit
for i in range(iter):
offset = i * 50
data = json.loads(requests.get(
f"https://api.opensea.io/api/v1/assets?order_direction=asc&offset={offset}"
f"&limit=50&collection={CollectionName}&format=json", headers=headers).content.decode())

if "assets" in data:
for asset in data["assets"]:
formatted_number = f"{int(asset['token_id']):04d}"

```
look like i setup right?

**Comment by NtekShadow** [comment]
wont be fixing anything. its not my projekt. good luck.

(https://status.opensea.io/)

**Comment by sj-dan** [comment]
Issue fixed by #11.
